CHANGELOG.md

```diff
@@ -30,6 +30,7 @@
 * Unless you specify the QGT implementation you wish to use with SR, we use an automatic heuristic based
   on your model and the solver to pick one. 
   This might affect SR performance. [#674](https://github.com/netket/netket/pull/674)
+* `netket.utils.group` provides utilities for geometrical and permutation groups. `Lattice` (and its specialisations like `Grid`) use these to automatically construct the space groups of lattices, as well as their character tables for generating wave functions with broken symmetry. [#724](https://github.com/netket/netket/pull/724)
```

Comment:
cc @femtobit 
Except rare cases (like `MPI`, or `configs`) I'd like to avoid having users reach into `netket.utils`.
It's more of an utility for us writing the inner netket than for end-users. 

I am seriously thinking that this could be it's own module.

--

In any case, @attila-i-szabo Do end-users needs to use functions from `netket.utils.group` to do something, or they are just fine with using the classes they are given and the methods on the class?

Reply:
You could argue that one only really needs `group` for constructing symmetry groups not provided out of the box. OTOH, people _do_ have to work with objects from `group` (such as specifically `PermutationGroup`, `Permutation`, ...).

The question is maybe: Do we consider "adding symmetries beyond what we provide" to be a end user thing?

Reply:
not for now.
Maybe in the future once the api is stabilised.
But not now.

So let's keep in in utils